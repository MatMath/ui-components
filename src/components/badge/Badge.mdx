---
name: Badge
menu: Components
---

import { Playground, Props } from 'docz';
import { Badge } from './Badge';
import '../../utility/playgroundStyles.css';

# Badge

A badge component

## Properties

<Props of={Badge} />

## Usage

### Basic usage

Examples below show a basic `Badge`

<Playground
  className={{preview: 'playground-base'}}
>
  <Badge text='Active' textColor='#FFFFFF' backgroundColor='#20B422'/>

</Playground>

<Playground
  className={{preview: 'playground-base'}}
>
  <Badge text='Pending' textColor='#0E353D' backgroundColor='#FDB5AF'/>

</Playground>
