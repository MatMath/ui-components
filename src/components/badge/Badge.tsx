import * as React from 'react';
import styles from './Badge.module.scss';

export interface BadgeProps {
  text: string;
  textColor: string;
  backgroundColor: string;
}

export const Badge = ({ text, textColor, backgroundColor }: BadgeProps) => {
  return (
    <div
      className={styles.badge}
      style={{
        color: `${textColor}`,
        backgroundColor: `${backgroundColor}`
      }}
    >
      {text}
    </div>
  );
};
