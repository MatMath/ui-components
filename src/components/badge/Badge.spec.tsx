import * as React from 'react';
import { render } from '@testing-library/react';
import { Badge } from './Badge';

describe('<Badge>', () => {
  describe('Global render', () => {
    it('should render default', () => {
      const { container } = render(
        <Badge text='Custom badge' textColor='white' backgroundColor='black' />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
