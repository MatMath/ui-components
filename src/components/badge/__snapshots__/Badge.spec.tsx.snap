// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge> Global render should render default 1`] = `
<div>
  <div
    class="badge"
    style="color: white; background-color: black;"
  >
    Custom badge
  </div>
</div>
`;
